From b5ef7872766518c0a0cb0ecabb29c5b2236d0def Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Mon, 17 Jun 2019 17:19:14 +0800
Subject: [PATCH 1/2] fbdev:mxsfb:invoke display_timings_release() to prevent
 memory leak

Use display_timings_release() to replace kfree() to prevent below
memory leak in mxsfb driver:

unreferenced object 0xa976c200 (size 128):
comm "kworker/1:1", pid 24, jiffies 4294937915 (age 281.840s)
hex dump (first 32 bytes):
80 61 8c 00 80 61 8c 00 80 61 8c 00 e0 01 00 00 .a...a...a......
e0 01 00 00 e0 01 00 00 08 00 00 00 08 00 00 00 ................
backtrace:
[<311bbb2b>] kmemleak_alloc+0x40/0x74
[<a97a5d3c>] kmem_cache_alloc_trace+0x1fc/0x2d4
[<f78878d0>] of_get_display_timings+0x1b4/0x204
[<379b07e1>] mxsfb_init_fbinfo_dt+0x1a4/0x2c8
[<3a23b8d7>] mxsfb_probe+0x2e4/0xd24
[<fbcba9bc>] platform_drv_probe+0x58/0xa4
[<7223258a>] driver_probe_device+0x240/0x308
[<e44060ff>] __device_attach_driver+0xa4/0xbc
[<0b7b416e>] bus_for_each_drv+0xb0/0xc4
[<2da12e51>] __device_attach+0xc4/0x138
[<433d5240>] device_initial_probe+0x1c/0x20
[<8dc18b30>] bus_probe_device+0x38/0x90
[<a127f59f>] deferred_probe_work_func+0x114/0x12c
[<7d7b4b19>] process_one_work+0x29c/0x480
[<b6f459c8>] process_scheduled_works+0x38/0x3c
[<9c0ba98c>] worker_thread+0x2b0/0x414

Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 drivers/video/fbdev/mxsfb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index b28c3b9c6b24..ab402d3cc60e 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -1410,7 +1410,7 @@ static int mxsfb_init_fbinfo_dt(struct fb_info *fb_info)
 	of_node_put(timings_np);
 put_display_node:
 	if (timings)
-		kfree(timings);
+		display_timings_release(timings);
 	of_node_put(display_np);
 	return ret;
 }
-- 
2.17.1

